services/web/server/src/simcore_service_webserver/studies_access.py

```diff
@@ -71,46 +77,40 @@ async def create_temporary_user(request: web.Request):
     email = username + "@guest-at-osparc.io"
     password = get_random_string(min_len=12)
 
-    # creates a user that is marked as ANONYMOUS. see update_user_as_guest
-    user = await db.create_user(
-        {
-            "name": username,
-            "email": email,
-            "password_hash": encrypt_password(password),
-            "status": ACTIVE,
-            "role": ANONYMOUS,
-            "created_ip": get_client_ip(request),
-        }
-    )
-
-    return user
-
-
-async def activate_as_guest_user(user: Dict, app: web.Application):
-    from .login.cfg import get_storage
-    from .login.handlers import GUEST, ANONYMOUS
-    from .resource_manager.websocket_manager import WebsocketRegistry
-
-    # Save time to allow user to be redirected back and not being deleted by GC
-    # SEE https://github.com/ITISFoundation/osparc-simcore/pull/1928#discussion_r517176479
-    EXTRA_TIME_TO_COMPLETE_REDIRECT = 3
-
-    if user.get("role") == ANONYMOUS:
-        db = get_storage(app)
-        username = user["name"]
+    lock_manager: Aioredlock = request.app[APP_CLIENT_REDIS_LOCK_KEY]
 
-        # creates an entry in the socket's registry to avoid garbage collector (GC) deleting it
-        # SEE https://github.com/ITISFoundation/osparc-simcore/issues/1853
+    try:
+        async with await lock_manager.lock(GC_EXECUTION_LOCK, lock_timeout=3):
+            user = await db.create_user(
+                {
+                    "name": username,
+                    "email": email,
+                    "password_hash": encrypt_password(password),
+                    "status": ACTIVE,
+                    "role": GUEST,
+                    "created_ip": get_client_ip(request),
+                }
+            )
+            # time-out lock to prevent GC from deleting this user until it acquires its first resources
+            await lock_manager.lock(
+                GUEST_USER_RC_LOCK_FORMAT.format(user_id=user["id"]),
+                lock_timeout=30,
+            )
```

Comment:
how do you know 3 seconds is enough/too much?

Reply:
and maybe a const value would be more verbose?

Reply:
done